src/Processors/QueryPlan/MergingAggregatedStep.cpp

```diff
@@ -152,6 +155,8 @@ void MergingAggregatedStep::describeActions(JSONBuilder::JSONMap & map) const
 void MergingAggregatedStep::updateOutputStream()
 {
     output_stream = createOutputStream(input_streams.front(), params.getHeader(input_streams.front().header, final), getDataStreamTraits());
+    if (is_order_overwritten)  /// overwrite order again
+        applyOrder(group_by_sort_description, overwritten_sort_scope);
```

Comment:
Let's add a comment that `createOutputStream` drops sort description because `preserves_sorting = false` .

Maybe flag `is_order_overwritten` is not needed and we can check `!group_by_sort_description.empty()`